backend/test/modules/doubles-tournament/core/application/BookTableAndPublishEventAfterCallingMatch.spec.ts

```diff
@@ -0,0 +1,187 @@
+
+    //Then
+    expect(commandResult.isSuccess()).toBeTruthy();
+    expect(doublesTournament.lastPublishedEvent()).toStrictEqual(
+      new MatchWasCalled({
+        occurredAt: currentTime,
+        tournamentId: tournamentId,
+        matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+        tableNumber: tableNumber,
+      }),
+    );
+
+    expect(executeSpy).toHaveBeenCalledTimes(2);
+
+    const bookFromAvailableTables = new BookTournamentTable(tournamentId, tableNumber);
+    expect(commandBus.execute).toHaveBeenLastCalledWith(bookFromAvailableTables);
+  });
+
+  //--------------------------------------------------------------------------------------------------------------------
+
+  it('Then publish event and execute BookTournamentTable command from different module | everything success', async () => {
+    //Given
+    const bookTournamentTableCommandHandler: CommandHandler<BookTournamentTable> = {
+      async execute(command: BookTournamentTable): Promise<CommandResult> {
+        return CommandResult.success();
+      },
+    };
+
+    const commandBus = new InMemoryCommandBus();
+    commandBus.registerHandler(BookTournamentTable, bookTournamentTableCommandHandler);
+    const executeSpy = jest.spyOn(commandBus, 'execute');
+
+    const doublesTournament = testDoublesTournamentsModule(currentTime, entityIdGen, commandBus);
+
+    //When
+    const callMatch = new CallMatch({
+      tournamentId: tournamentId,
+      matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+      tableNumber: tableNumber,
+    });
+    const commandResultFromCallingMatch = await doublesTournament.executeCommand(callMatch);
+
+    const bookTable = new BookTournamentTable(tournamentId, tableNumber);
+    const commandResultFromBookingTable = await doublesTournament.executeCommand(bookTable);
+
+    //Then
+    expect(commandResultFromCallingMatch.isSuccess()).toBeTruthy();
+    expect(commandResultFromBookingTable.isSuccess()).toBeTruthy();
+
+    expect(doublesTournament.lastPublishedEvent()).toStrictEqual(
+      new MatchWasCalled({
+        occurredAt: currentTime,
+        tournamentId: tournamentId,
+        matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+        tableNumber: tableNumber,
+      }),
+    );
+
+    expect(executeSpy).toHaveBeenCalledTimes(3);
+
+    const bookFromAvailableTables = new BookTournamentTable(tournamentId, tableNumber);
+    expect(commandBus.execute).toHaveBeenLastCalledWith(bookFromAvailableTables);
+  });
+
+  it('Then publish event and execute BookTournamentTable command from different module | table has been already booked', async () => {
+    //Given
+    const bookTournamentTableCommandHandler: CommandHandler<BookTournamentTable> = {
+      async execute(command: BookTournamentTable): Promise<CommandResult> {
+        return CommandResult.failureDueTo(new Error('Table has been already booked!'));
```

Comment:
Bardzo ładnie rozkminiłeś jak to zamockować :)

Reply:
mocne słowa z tym rozkminianiem :-P
Sam takie coś zrobiłeś w InMemoryCommandBus.spec.ts :-)

Reply:
Tam jest troche inny przypadek. Ooo, ale dobrze ze zglosiles sie do odpowiedzi. To co to tam robi :) ? I jak to wplywa na dzialanie aplikacji i kodu command handlerów.

Reply:
No w tym InMemoryCommandBus są wypisane przypadki testowe do komendy startowania turnieju m.in jak wynikiem execute nie będzie success tylko failure (co właśnie potrzebowałem zrobić w swoim command handlerze). Są też przypadki do tego, żeby nie móc zarejestrować innego (kolejnego) caommandHandlera dla startowania turnieju i też **na końcu sprawdzamy, że nie uda się zrobić execute na komendzie, która nie ma zarejestrowanego commandHandlera** 

Faktycznie po dobrze opisanych testach można sporo wynieść 👍